### PR TITLE
Add advanced_backup_setting

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Add advanced_backup_setting

## why
* Backs up a specific resource like EC2. Only works for windows at the moment.

## references
* Closes https://github.com/cloudposse/terraform-aws-backup/issues/35
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/backup_plan#advanced_backup_setting

### CODE REVIEW

**Comment by nightspotlight** [comment]
since this option is only available for Windows VSS (https://docs.aws.amazon.com/aws-backup/latest/devguide/API_AdvancedBackupSetting.html), I'd prefer a simple switch rather than an arbitrary data structure.

for example, if say `var.windows_vss_backup` is true, then we enable this block:
```hcl
advanced_backup_setting {
  backup_options = {
    WindowsVSS = "enabled"
  }
  resource_type = "EC2"
}
```

so maybe something like this instead:
```hcl
variable "windows_vss_backup" {
  type = bool
  default = false
}
```
```hcl
dynamic "advanced_backup_setting" {
  for_each = var.windows_vss_backup ? [true] : []
  content {
    backup_options = {
      WindowsVSS = "enabled"
    }
    resource_type = "EC2"
  }
}
```

File: main.tf [hunk 1]
```diff
@@ -98,6 +98,15 @@ resource "aws_backup_plan" "default" {
     }
   }
 
+  dynamic "advanced_backup_setting" {
+    for_each = var.advanced_backup_setting != null ? [true] : []
+
+    content {
+      resource_type  = lookup(var.advanced_backup_setting, "resource_type")
+      backup_options = lookup(var.advanced_backup_setting, "backup_options")
```
Comment (nightspotlight) [inline main.tf:106]: using lookup function without the third argument is deprecated (https://www.terraform.io/language/functions/lookup)

**Comment by Benbentwo** [comment]
Closing this as this should be resolved by #63

**Pull request closed without merging**